modules/javafx.graphics/src/main/java/javafx/scene/Scene.java

```diff
@@ -1285,71 +1285,27 @@ static WritableImage doSnapshot(Scene scene,
             Node root, BaseTransform transform, boolean depthBuffer,
             Paint fill, Camera camera, WritableImage wimg) {
 
+        Toolkit tk = Toolkit.getToolkit();
+        Toolkit.ImageRenderingContext context = new Toolkit.ImageRenderingContext();
+
         int xMin = (int)Math.floor(x);
         int yMin = (int)Math.floor(y);
-        int width;
-        int height;
+        int xMax = (int)Math.ceil(x + w);
+        int yMax = (int)Math.ceil(y + h);
+        int width = Math.max(xMax - xMin, 1);
+        int height = Math.max(yMax - yMin, 1);
```

Comment:
The changes here do not need to be reverted, this is still extra calculation.

Reply:
Well spotted. I've re-applied that change.